dependencies.yaml

```diff
@@ -165,6 +168,9 @@ dependencies:
       - output_types: [conda, pyproject, requirements]
         packages:
           - numpy >=1.23,<3.0.0a0
+      - output_types: conda
+        packages:
+          - nccl
 
```

Comment:
```suggestion
```

I think this should be removed. It could cover up mistakes in `legate-dataframe`'s packaging. We should rely on the `legate-dataframe` conda packages' `run:` dependency on `nccl` to pull this in.